### PR TITLE
Tooltip 2

### PR DESCRIPTION
**Pull request by duncanspumpkin**
A couple of little refactors to the tooltips to make it a bit simpler.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
This should fix the introduced issues with the tooltips now. I did a search for all tooltips that had parameters and ensured they were taken care of so I think that has caught all of the issues.

File: src/openrct2-ui/windows/EditorInventionsList.cpp [hunk 1]
```diff
@@ -714,12 +714,12 @@ static void window_editor_inventions_list_drag_open(ResearchItem* researchItem)
     window_editor_inventions_list_drag_widgets[0].right = stringWidth;
 
     w = window_create(
-        ScreenCoordsXY(gTooltipCursorX - (stringWidth / 2), gTooltipCursorY - 7), stringWidth, 14,
+        ScreenCoordsXY(gTooltipCursor.x - (stringWidth / 2), gTooltipCursor.y - 7), stringWidth, 14,
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorInventionsList.cpp:717]:
```suggestion
        gTooltipCursor - ScreenCoordsXY{ stringWidth / 2, 7 }, stringWidth, 14,
```

File: src/openrct2-ui/windows/Tooltip.cpp [hunk 2]
```diff
@@ -47,30 +47,35 @@ void window_tooltip_reset(const ScreenCoordsXY& screenCoords)
...
+    rct_window* w;
+    int32_t width, height;
+
+    w = window_find_by_class(WC_ERROR);
+    if (w != nullptr)
+        return;
+
+    int32_t textWidth = FormatTextForTooltip(message);
+    width = textWidth + 3;
+    height = ((_tooltipNumLines + 1) * font_get_line_height(FONT_SPRITE_BASE_MEDIUM)) + 4;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Tooltip.cpp]:
```suggestion
    auto* w = window_find_by_class(WC_ERROR);
    if (w != nullptr)
        return;

    int32_t textWidth = FormatTextForTooltip(message);
    int32_t width = textWidth + 3;
    int32_t height = ((_tooltipNumLines + 1) * font_get_line_height(FONT_SPRITE_BASE_MEDIUM)) + 4;
```

File: src/openrct2-ui/windows/Ride.cpp [hunk 3]
```diff
@@ -2777,6 +2779,58 @@ static void window_ride_vehicle_update(rct_window* w)
...
+        return { STR_NONE, {} };
+
+    switch (widgetIndex)
+    {
+        case WIDX_VEHICLE_TRAINS:
+        case WIDX_VEHICLE_TRAINS_DECREASE:
+        case WIDX_VEHICLE_TRAINS_INCREASE:
+        {
+            auto ft = Formatter();
+            ft.Increment(12);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Ride.cpp:2796]: What's up with these increments that didn't exit before? There isn't any data prior nor any rewind later, so they seems a bit confusing (to me, at least)

File: src/openrct2-ui/windows/Ride.cpp [hunk 4]
```diff
@@ -2777,6 +2779,58 @@ static void window_ride_vehicle_update(rct_window* w)
...
+            rct_string_id stringId = RideComponentNames[vehicleType].count;
+            if (ride->max_trains > 1)
+            {
+                stringId = RideComponentNames[vehicleType].count_plural;
+            }
+            ft.Add<rct_string_id>(stringId);
+            ft.Add<uint16_t>(ride->max_trains);
+            return { fallback, ft };
+        }
+        break;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Ride.cpp:2808]:
```suggestion
```

File: src/openrct2-ui/windows/Ride.cpp [hunk 5]
```diff
@@ -2777,6 +2779,58 @@ static void window_ride_vehicle_update(rct_window* w)
...
+
+            rct_string_id stringId = RideComponentNames[RIDE_COMPONENT_TYPE_CAR].singular;
+            if ((ride->min_max_cars_per_train & 0xF) - rideEntry->zero_cars > 1)
+            {
+                stringId = RideComponentNames[RIDE_COMPONENT_TYPE_CAR].plural;
+            }
+            ft.Add<rct_string_id>(stringId);
+            return { fallback, ft };
+        }
+        break;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Ride.cpp:2829]:
```suggestion
```

**Comment by tupaschoal** [comment]
I've fixed the build failure myself, but there are still some comments and suggestions there.

File: src/openrct2-ui/windows/Ride.cpp (hunk 3 above)
Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Ride.cpp:2796]: The string id associated with this needs the args in those positions. If you look at the chain of adds that this was removed from you will see where the number comes from

**Pull request merged**